src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -512,6 +513,175 @@ pass::EliminateConcat::EliminateConcat() {
+        const auto concat = ov::as_type_ptr<ov::op::v0::Concat>(pattern_map.at(pattern_concat));
+
+        const auto concat_axis =
+            ov::util::normalize(concat->get_axis(), concat->get_output_partial_shape(0).rank().get_length());
+
+        const auto concat_users = concat->get_users();
+        if (concat_users.size() == 1)
+            return false;
+        auto concat_inputs = concat->inputs();
+
+        std::vector<std::tuple<std::shared_ptr<Node>, int64_t, int64_t>> slice_out_index_in_concat;
+        for (const auto& user : concat_users) {
+            if (ov::is_type<ov::op::v1::StridedSlice>(user)) {
+                auto strided_slice_node = ov::as_type_ptr<ov::op::v1::StridedSlice>(user);
+                if (!strided_slice_node) {
+                    return false;
+                }
+                // check that all values of the mask is equal 0
+                auto check_mask = [](const std::vector<int64_t>& mask_to_check) {
+                    auto it = std::find_if(mask_to_check.begin(), mask_to_check.end(), [](const int64_t& value) {
+                        return value != 0;
+                    });
+                    if (mask_to_check.empty() || it == mask_to_check.end()) {
+                        return true;
+                    }
+                    return false;
+                };
+                // check that we won't do change dimention rank
+                if (!check_mask(strided_slice_node->get_shrink_axis_mask()) ||
+                    !check_mask(strided_slice_node->get_new_axis_mask()) ||
+                    !check_mask(strided_slice_node->get_ellipsis_mask())) {
+                    return false;
+                }
+
+                auto begin_node = strided_slice_node->get_input_node_shared_ptr(1);
+                const auto& begin_constant_node = ov::util::get_constant_from_source(begin_node);
+                if (begin_constant_node == nullptr)
+                    return false;
+                auto begin_values = begin_constant_node->cast_vector<int64_t>();
+
+                auto end_node = strided_slice_node->get_input_node_shared_ptr(2);
+                const auto& end_constant_node = ov::util::get_constant_from_source(end_node);
+                if (end_constant_node == nullptr)
+                    return false;
+                auto end_values = end_constant_node->cast_vector<int64_t>();
+
+                slice_out_index_in_concat.push_back(
+                    std::make_tuple(strided_slice_node, begin_values[concat_axis], end_values[concat_axis] - 1));
+            } else {
+                return false;
+            }
+        }
+        if (slice_out_index_in_concat.size() == 1)
+            return false;
+
+        uint64_t start_index = 0;
+        std::vector<std::tuple<std::shared_ptr<Node>, int64_t, int64_t>> in_index_in_concat;
+        for (auto& concat_in : concat_inputs) {
+            auto tmp_index = start_index + concat_in.get_shape()[concat_axis] - 1;
```

Comment:
the input shape might be dynamic here
`concat_in.get_shape()` throws an exception in this case.
we need to use `.get_partial_shape()[concat_axis].get_length()`

Reply:
Yes, u are right. For now, the dynamic shape doesn't be supported, which will be done in next step. So I will check the concat node is dynamic or not in the beginning.